src/utilities/compareVersions.js

```diff
@@ -0,0 +1,69 @@
+import { Client as BugsnagClient } from 'bugsnag-react-native';
+const bugsnagClient = new BugsnagClient();
+
+export function compareVersions(versionOne, versionTwo) {
+  try {
+    const result = versionToInteger(versionOne) - versionToInteger(versionTwo);
+    if (result > 0) return 1;
+    if (result < 0) return -1;
+    return 0;
+  } catch (error) {
+      // Errors thrown in migration method in dataMigration.js do not trickle down to bugsnag
+      // so will need manually notify bug snag.
+    error.message = `error verifying versions, versionOne: ${versionOne}, versionTwo ${versionTwo}`;
+    bugsnagClient.notify(error);
+  }
+  return 0;
+}
+// versionToInteger logic is the same as ./android/app/build.gradle
+function versionToInteger(version) {
+  const [majorMinorPatch, provisional] = version.split('-');
+  const [major, minor, patch] = majorMinorPatch.split('.');
+  return Number(major) * 10000000 +
+         Number(minor) * 100000 +
+         Number(patch) * 100 +
+         Number(!provisional ? '99' : provisional.match(/[0-9]+/)[0]);
+         // Regex /[0-9]+/ matches only numbers and at least on digit number
+}
+
+export function test() {
```

Comment:
Would be better in a Jest test but don't bother. We don't have any level of automated tests sadly.